src/services/kratos/auth-phone-no-password.ts

```diff
@@ -173,6 +173,40 @@ export const AuthWithPhonePasswordlessService = (): IAuthWithPhonePasswordlessSe
     }
   }
 
+  const createIdentityWithDeviceId = async ({
+    phone,
+    deviceId,
+  }: {
+    phone: PhoneNumber
+    deviceId: UserId
+  }): Promise<CreateKratosUserForPhoneNoPasswordSchemaResponse | KratosError> => {
+    // need to use phone and deviceId as the identifier for the kratos webhook callback to work
+    //  in /kratos/registration
+    const traits = { phone, deviceId }
+    const method = "password"
+    try {
+      const flow = await kratosPublic.createNativeRegistrationFlow()
+      const result = await kratosPublic.updateRegistrationFlow({
+        flow: flow.data.id,
+        updateRegistrationFlowBody: {
+          traits,
+          method,
+          password,
```

Comment:
it seems you could also use `transient_payload` instead of `deviceId`. one of the benefits is that the kratos entity would not be bloated with deviceId because it won't be use anymore, and this would simplify any further migration of the kratos entity in the future.

deviceId should still be stored somewhere, but this could be in the User mongodb collection. I think this would be a better place than here, because it's just metadata. and adding a `traits` doesn't feels the ideal place for metadata as this is directly related to how our authentication works.


documentation of `transient_payload`:
    /**
     * Transient data to pass along to any webhooks
     * @type {object}
     * @memberof UpdateRegistrationFlowWithOidcMethod
     */
    'transient_payload'?: object;